src/app/services/alert/alert.service.ts

```diff
@@ -1,28 +1,34 @@
 import { Injectable } from '@angular/core';
 import { AlertController } from '@ionic/angular';
+import { TranslateService } from '@ngx-translate/core';
+
 @Injectable({
   providedIn: 'root'
 })
 export class AlertService {
   alert: any;
 
-  constructor(private alertController: AlertController) {}
+  constructor(private alertController: AlertController,private translate:TranslateService) { }
 
   async presentAlert(header: string, message: string, buttons: { text: string, cssClass?: string, role?: string, handler?: () => void }[]) {
-    this.alert = await this.alertController.create({
-      header,
-      message,
-      buttons: buttons.map(button => ({
-        text: button.text,
+    this.translate.get([message,header,...buttons.map(b => b.text)]).subscribe(async (data:any) => {
```

Comment:
After translation can we assign the keys directly instead of assigning it here and using it in line 24, 25, 26